### PR TITLE
Moves the `~/.spack` directory to /software

### PR DESCRIPTION
**Pull request by dipietrantonio**
This will avoid hitting the /home quota, and will allow the user to have a different spack cache for each project. But most importantly, it will allow us to parametrise the user configuration path (a.k.a ~/.spack) based on the software stack version.

Opening this merge request for you to review the strategy, do not merge yet, more testing is needed.

### CODE REVIEW

**Comment by dipietrantonio** [comment]
I let spack run some installations, and a `~/.spack` directory is still being created to save some kind of report
```
cdipietrantonio@setonix-01:~> ls -lah ~ | grep spack
drwxr-xr-x    3 cdipietrantonio pawsey0001      4.0K Jan 11 16:33 .spack
cdipietrantonio@setonix-01:~> ls -lah .spack/reports/
total 12K
drwxr-xr-x 3 cdipietrantonio pawsey0001 4.0K Jan 11 16:33 .
drwxr-xr-x 3 cdipietrantonio pawsey0001 4.0K Jan 11 16:33 ..
drwxr-xr-x 2 cdipietrantonio pawsey0001 4.0K Jan 11 16:33 junit
cdipietrantonio@setonix-01:~> ls -lah .spack/
total 16K
drwxr-xr-x  3 cdipietrantonio pawsey0001      4.0K Jan 11 16:33 .
drwx------ 19 cdipietrantonio cdipietrantonio 8.0K Jan 11 16:43 ..
drwxr-xr-x  3 cdipietrantonio pawsey0001      4.0K Jan 11 16:33 reports
cdipietrantonio@setonix-01:~> 
```
However, our custom spack user config location works for settings and bootstrap files
```
cdipietrantonio@setonix-01:~> ls /software/projects/pawsey0001/cdipietrantonio/setonix/.spack_user_config/
bootstrap  config.yaml	modules.yaml  upstreams.yaml
```

I just need to figure where those reports are generated.

**Comment by dipietrantonio** [comment]
My tests show that these changes work, I will merge once the reviewers approve the changes.

**Review by pelahi — APPROVED** [review]
Looks good

**Review by ddeeptimahanti — APPROVED** [review]

**Pull request merged**
